core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -2205,12 +2190,12 @@ private[spark] class DAGScheduler(
           val message = s"Stage failed because barrier task $task finished unsuccessfully.\n" +
             failure.toErrorString
           try {
-            // killAllTaskAttempts will fail if a SchedulerBackend does not implement killTask.
+            // cancelTasks will fail if a SchedulerBackend does not implement killTask.
             val reason = s"Task $task from barrier stage $failedStage (${failedStage.name}) " +
               "failed."
             val job = jobIdToActiveJob.get(failedStage.firstJobId)
             val shouldInterrupt = job.exists(j => shouldInterruptTaskThread(j))
-            taskScheduler.killAllTaskAttempts(stageId, shouldInterrupt, reason)
+            taskScheduler.cancelTasks(stageId, shouldInterrupt, reason)
```

Comment:
shall we also abort the stage here?

Reply:
It is changed from `killAllTaskAttempts`, which doesn't abort the stage. So we don't abort here either.